Automattic/wp-calypso
# Correctly report eligibility page view and report clicks for plugin/theme upload buttons

`<EligibilityWarnings />` component that is displayed on a standalone page as well as in places such as `/plugins` and `/themes` logs a page view with hardcoded URL. This PR fixes that so only the standalone page logs a page view. Also, it adds additional events when user clicks on "Upload theme" or "Upload plugin" buttons.

Test plan:
1. Navigate to a free site in Calypso
1. Go to `/themes` and confirm in network panel of dev tools that only a page view for `/themes` is recorded
1. Go to `/plugins` and confirm in network panel of dev tools that only a page view for `/plugins` is recorded
1. Go to `/plugins/tinymce-advanced` and confirm in network panel of dev tools that only a page view for `/plugins/tinymce-advanced` is recorded
1. Go to `/plugins/tinymce-advanced/eligibility/` and confirm in network panel of dev tools that only a page view for `/plugins/tinymce-advanced/eligibility/` is logged
1. Go to `/plugins` and click on "Upload plugin" button, confirm in devtools that `calypso_click_plugin_upload` event was logged
1. Go to `/plugins/manage` and click on "Upload plugin" button, confirm in devtools that `calypso_click_plugin_upload` event was logged
1. Go to `/themes` and click on "Upload theme" button, confirm in devtools that `calypso_click_theme_upload` event was logged
